src/schema/resolvers.js

```diff
@@ -572,14 +572,29 @@ module.exports = {
       return tx;
     },
 
-    finalizeResult: async (root, data, { db: { Transactions } }) => {
+    finalizeResult: async (root, data, { db: { Oracles, Transactions } }) => {
       const {
         version,
         topicAddress,
         oracleAddress,
         senderAddress,
       } = data;
 
+      // Fetch oracle to get the finalized result
+      const oracle = await Oracles.findOne({ address: oracleAddress }, { options: 1, optionIdxs: 1 });
+      let winningIndex;
+      if (!oracle) {
+        logger.error(`Could not find Oracle ${address} in DB.`);
+        throw new Error(`Could not find Oracle ${address} in DB.`);
+      } else {
+        for (let i = 0; i < oracle.options.length; i++) {
```

Comment:
nit: add a comment to explain comparison between `options` & `optionIdxs`

Reply:
done